### PR TITLE
Do not declare external runtime libraries in POM file

### PR DESCRIPTION
**Pull request by kkocel**
Since the shadow plugin bundles all libraries used by axion it's no longer necessary to declare them in POM file.

This PR makes axion self-contained and not declaring not used libraries (since they're shadowed).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -114,6 +114,10 @@ shadowJar {
     archiveClassifier.set('')
 }
 
+components.java.withVariantsFromConfiguration(configurations.runtimeElements) {
+    skip()
```
Comment (bgalek) [inline build.gradle:118]: is this pom related?

**Review by bgalek — CHANGES_REQUESTED** [review]

**Comment by bgalek** [comment]
this can be a problem, some libraries load classes via reflection so shadowing them can break things

File: build.gradle [hunk 2]
```diff
@@ -83,8 +84,7 @@ dependencies {
         exclude group: 'org.eclipse.jgit', module: 'org.eclipse.jgit'
     }
 
-    testImplementation group: 'org.testcontainers', name: 'spock', version: '1.15.3'
-    testImplementation group: 'org.codehaus.groovy', name: 'groovy-all', version: '3.0.7'
+    testImplementation group: 'org.testcontainers', name: 'spock', version: '1.16.0'
```
[the remark this answers is not in the document]
Comment (kkocel) [reply build.gradle:87]: I'm not sure if I follow.
Please take a look at the following test: https://github.com/allegro/axion-release-plugin/blob/b6e82a72d85293f57fd7141bbc2a07007291bb32/src/integration/groovy/pl/allegro/tech/build/axion/release/RemoteRejectionTest.groovy

What it does:
1. builds a custom Docker image from a `Dockerfile`
2. runs it
3. checks code behavior against a real server

If there are users without Docker I can pass a flag to this test that will ignore it if Docker is not present at host machine. Seems a good trade-off for me (test will be run on CI)

File: build.gradle (hunk 1 above)
Comment (kkocel) [reply build.gradle:118]: yes, It clears output pom (and Gradle module metadata) from runtime dependencies

Comment (kkocel) [reply build.gradle:118]: Before:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <!-- This module was also published with a richer model, Gradle metadata,  -->
  <!-- which should be used instead. Do not delete the following line which  -->
  <!-- is to indicate to Gradle or any Gradle module metadata file consumer  -->
  <!-- that they should prefer consuming it instead. -->
  <!-- do_not_remove: published-with-gradle-metadata -->
  <modelVersion>4.0.0</modelVersion>
  <groupId>pl.allegro.tech.build</groupId>
  <artifactId>axion-release-plugin</artifactId>
  <version>1.13.4-SNAPSHOT</version>
  <dependencies>
    <dependency>
      <groupId>org.eclipse.jgit</groupId>
      <artifactId>org.eclipse.jgit</artifactId>
      <version>5.12.0.202106070339-r</version>
      <scope>runtime</scope>
    </dependency>
    <dependency>
      <groupId>org.eclipse.jgit</groupId>
      <artifactId>org.eclipse.jgit.ssh.jsch</artifactId>
      <version>5.12.0.202106070339-r</version>
      <scope>runtime</scope>
    </dependency>
    <dependency>
      <groupId>com.jcraft</groupId>
      <artifactId>jsch</artifactId>
      <version>0.1.55</version>
      <scope>runtime</scope>
    </dependency>
    <dependency>
      <groupId>com.jcraft</groupId>
      <artifactId>jsch.agentproxy.core</artifactId>
      <version>0.0.9</version>
      <scope>runtime</scope>
    </dependency>
    <dependency>
      <groupId>com.jcraft</groupId>
      <artifactId>jsch.agentproxy.jsch</artifactId>
      <version>0.0.9</version>
      <scope>runtime</scope>
    </dependency>
    <dependency>
      <groupId>com.jcraft</groupId>
      <artifactId>jsch.agentproxy.sshagent</artifactId>
      <version>0.0.9</version>
      <scope>runtime</scope>
    </dependency>
    <dependency>
      <groupId>com.jcraft</groupId>
      <artifactId>jsch.agentproxy.pageant</artifactId>
      <version>0.0.9</version>
      <scope>runtime</scope>
    </dependency>
    <dependency>
      <groupId>com.jcraft</groupId>
      <artifactId>jsch.agentproxy.usocket-jna</artifactId>
      <version>0.0.9</version>
      <scope>runtime</scope>
    </dependency>
    <dependency>
      <groupId>com.jcraft</groupId>
      <artifactId>jsch.agentproxy.usocket-nc</artifactId>
      <version>0.0.9</version>
      <scope>runtime</scope>
    </dependency>
    <dependency>
      <groupId>com.github.zafarkhaja</groupId>
      <artifactId>java-semver</artifactId>
      <version>0.9.0</version>
      <scope>runtime</scope>
    </dependency>
    <dependency>
      <groupId>org.eclipse.jgit</groupId>
      <artifactId>org.eclipse.jgit.ui</artifactId>
      <version>5.12.0.202106070339-r</version>
      <scope>runtime</scope>
    </dependency>
    <dependency>
      <groupId>org.eclipse.jgit</groupId>
      <artifactId>org.eclipse.jgit.gpg.bc</artifactId>
      <version>5.12.0.202106070339-r</version>
      <scope>runtime</scope>
    </dependency>
  </dependencies>
</project>
```
After:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <!-- This module was also published with a richer model, Gradle metadata,  -->
  <!-- which should be used instead. Do not delete the following line which  -->
  <!-- is to indicate to Gradle or any Gradle module metadata file consumer  -->
  <!-- that they should prefer consuming it instead. -->
  <!-- do_not_remove: published-with-gradle-metadata -->
  <modelVersion>4.0.0</modelVersion>
  <groupId>pl.allegro.tech.build</groupId>
  <artifactId>axion-release-plugin</artifactId>
  <version>1.13.4-SNAPSHOT</version>
</project>
```

File: build.gradle (hunk 2 above)
[the remark this answers is not in the document]
Comment (bgalek) [reply build.gradle:87]: for integrations tests it's ok, but when you remove groovy-all+spck dependency you won't be able to run unit tests without docker, right?

**Review by bgalek — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 2 above)
[the remark this answers is not in the document]
Comment (kkocel) [reply build.gradle:87]: `groovy-all` is provided by Gradle itself, I added explicit declaration here: https://github.com/allegro/axion-release-plugin/blame/24cecc613af9cca43305c8f3265c28e2de3c4bc6/build.gradle#L88 
but the explicit declaration of groovy is error-prone. Consider the following future situation:

1. You update Gradle to a newer version, 
2. Gradle comes with a new Groovy version
3. Whoops - you can't build axion because of groovy version conflicts 

That's why it's better to have implicit groovy dependency instead of an explicit one.

So to summarize changes in lines `86-89`: 
I bumped [tesctontainers](https://www.testcontainers.org/test_framework_integration/spock/) library (used by `RemoteRejectionTest.groovy`)
I removed the explicit groovy dependency declaration.

[the remark this answers is not in the document]
Comment (bgalek) [reply build.gradle:87]: ok, i mixed spock-core and test-containers spock
groovy-all from gradle seems ok, since this is a plugin
